src/tilda.c

```diff
@@ -746,11 +764,24 @@ int main (int argc, char *argv[])
         if (atol (getenv ("VTE_PROFILE_MEMORY")) != 0)
             g_mem_set_vtable (glib_mem_profiler_table);
 #endif
+    /* Parse the command line */
+    config_file = NULL;
+    need_wizard = parse_cli (argc, argv, &config_file);
+
+    if (config_file != NULL) {
+        if (!g_file_test (config_file, G_FILE_TEST_EXISTS)) {
+            g_printerr (_("Specified config file '%s' does not exist. Reverting to default path.\n"),
+                    config_file);
+            config_file = NULL;
+        }
+    }
+    if (config_file == NULL) {
+        config_file = get_config_file_name (lock.instance);
+    }
+
     /* Start up the configuration system */
     gint config_init_result = config_init (config_file);
 
-    /* Parse the command line */
-    need_wizard = parse_cli (argc, argv);
```

Comment:
I think you'll want to split it into two separate functions - `parse_cli` and `setup_config_from_cli_opts` where the after `parse_cli` you can check if a `config_file` was passed to Tilda or not before reading the config.  

e.g. https://github.com/pik/tilda/commit/04a33aaef4cc1d9ea72a18ec1bcd817fa9a23eb9#diff-78cb9b2d17dc8fc2e17756c87a36e752R447